### DISCUSSION TITLE
Code block background color

### DISCUSSION THREAD

**Discussion by purashy** [TriliumNext/Trilium#4503] [Q&A] [answered]
Hello, I have a weird question here.

Let me first start by what I have, and then with what I want.

**What I have at the moment:**

[Midnight theme](https://github.com/ttytm/trilium-midnight-theme) + [Syntax highlight](https://github.com/antoniotejada/Trilium-SyntaxHighlightWidget).

![image](https://github.com/zadam/trilium/assets/4239073/f76e74d2-8259-4d86-997b-9f300106f968)

Works great! And I really love the theme and everything.

**What I want:**

However, I want to change the codeblock background. It is a tad bit too grey for my taste.

The theme (Midnight) config is unchanged. The only part that I changed is at the top of the `SyntaxHighlightWidget.js` (These changes are purely for the syntax highlighting/changing the colors a bit):

```js
const TEMPLATE = `
<div style="padding: 10px; border-top: 1px solid var(--main-border-color); contain: none;">
<style>
pre code.hljs {
    display: block;
    overflow-x: auto;
    padding: 1em
}

code.hljs {
    padding: 3px 5px
}

.hljs-meta,
.hljs-comment {
  color: #999999;
}

/* Red */
/*INFO: This keyword, HTML elements, Regex group symbol, CSS units, Terminal Red */
.hljs-tag,
.hljs-doctag,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-template-tag,
.hljs-selector-pseudo,
.hljs-selector-attr,
.hljs-variable.language_,
.hljs-deletion {
  color: #f7768e;
}

/*Orange */
/*INFO: Number and Boolean constants, Language support constants */
.hljs-variable,
.hljs-template-variable,
.hljs-number,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-link {
  color: #ff9e64;
}


/*  Yellow */
/* INFO:  	Function parameters, Regex character sets, Terminal Yellow */
.hljs-built_in, 
.hljs-attribute {
  color: #e0af68;
}
/* cyan */
/* INFO: Language support functions, CSS HTML elements */
.hljs-selector-tag {
  color: #2ac3de;
}

/* light blue */
/* INFO: Object properties, Regex quantifiers and flags, Markdown headings, Terminal Cyan, Markdown code, Import/export keywords */
.hljs-keyword,
  .hljs-title.function_,
.hljs-title,
.hljs-title.class_,
.hljs-title.class_.inherited__,
.hljs-subst,
.hljs-property {color: #7dcfff;}

/*Green*/
/* INFO: Object literal keys, Markdown links, Terminal Green */
.hljs-selector-tag { color: #73daca;}


/*Green(er) */
/* INFO: Strings, CSS class names */
.hljs-quote,
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
  color: #9ece6a;
}

/* Blue */
/* INFO:  	Function names, CSS property names, Terminal Blue */
.hljs-code,
.hljs-formula,
.hljs-section {
  color: #7aa2f7;
}



/* Magenta */
/*INFO: Control Keywords, Storage Types, Regex symbols and operators, HTML Attributes, Terminal Magenta */
.hljs-name,
.hljs-keyword,
.hljs-operator,
.hljs-keyword,
.hljs-char.escape_,
.hljs-attr {
  color: #69f;
}

/* white*/
/* INFO: Variables, Class names, Terminal White */
.hljs-punctuation {color: #c0caf5}

.hljs {
  background: #9aa5ce;
  color: #9aa5ce;
}

.hljs-emphasis {
  font-style: italic;
}

.hljs-strong {
  font-weight: bold;
}
</style>
</div>`;
```

So to summarize, I want to change the Code block color from grey to a little bit darker color (to match the theme a little bit). Is this doable? I was unable to find anything about it.

Thank you in advance for the trouble 🙏

**Comment by zerebos** [comment]
It's something that wasn't really covered in midnight, you can add this to the bottom of the midnight css and then reload (ctrl+r)

```css
.ck-content pre {
    background: var(--left-pane-background-color);
    border: 1px solid var(--launcher-pane-background-color);
    --ck-color-code-block-label-background: var(--launcher-pane-background-color);
}
```

**Reply by purashy** [reply]
Added to the buttom of Midnight.css.
Reloaded (ctrl + r).
No changes however.

![image](https://github.com/zadam/trilium/assets/4239073/e029e5f4-8b8c-4564-b233-93839aec91bf)

**Edit:** Could have sworn this was a different color @rauenzi so maybe it did change but it only changed for this (right corner of codeblocks):

![image](https://github.com/zadam/trilium/assets/4239073/2115489f-53bb-4ca4-b432-623f850b81b4)

**Reply by zerebos** [reply, marked as answer]
Hmm I tested this with the syntax highlighting widget and the midnight theme on Trilium 0.62.3 and it worked just fine. You can try adding `!important` to the end of the first two rules like this:

```css
.ck-content pre {
    background: var(--left-pane-background-color)!important;
    border: 1px solid var(--launcher-pane-background-color)!important;
    --ck-color-code-block-label-background: var(--launcher-pane-background-color);
}
```

You have the latest unmodified version of midnight right?

**Reply by purashy** [reply]
I do indeed. Latest unmodified version.

Thank you, this code fixed it!

```css
.ck-content pre {
    background: var(--left-pane-background-color)!important;
    border: 1px solid var(--launcher-pane-background-color)!important;
    --ck-color-code-block-label-background: var(--launcher-pane-background-color);
}
```

![image](https://github.com/zadam/trilium/assets/4239073/de5f7476-a5da-4104-aff6-deacf720cd5c)

So much cleaner!!!! <3
